ragchain-structure/benchmark/dataset-evaluator/beir.md

```diff
@@ -0,0 +1,63 @@
+# Overview
+
+BEIR is a heterogeneous benchmark that has been built from 18 diverse datasets representing 9 information retrieval tasks.
+Using project, we use 6 datasets, FEVER, FIQA, HOTPOTQA, QUORA, SCIDOCS, and SCIFACT.
+More information about beir dataset, please visit link!
+
+[Github](https://github.com/beir-cellar/beir?tab=readme-ov-file)<br>
+[Paper](https://openreview.net/forum?id=wCu6T5xFjeJ)<br>
+[HuggningFace](https://huggingface.co/BeIR)
+
+
+Our BEIR benchmark structure is like below picture.
+Basic preprocessing process is same at 6 benchmark dataset,
+So we construct preprocess process, ingest, evaluation code are in BaseBeirEvaluator class.
+Each evaluator class corresponding to the benchmark datasets
+(FEVER, FIQA, HOTPOTQA, QUORA, SCIDOCS, and SCIFACT) inherits from the BaseBeirEvaluator.
+
+![beir-structure.png](../../../.gitbook/assets/beir-structure.png)
+
+Also, we removed non-relevant score passages at qrels.<br>
+The significance of this dataset is diverse tasks, diverse domains, task difficulties, and diverse annotation strategies
+in each dataset.
+
+Beir test code is little different from other benchmark.
+```Python
+bm25_path, pickle_path = generate_path('fever')
+retrieval, db, llm, pipeline = set_beir_evaluator(bm25_path, pickle_path)
+evaluator = <Beir evaluator>(run_pipeline=pipeline, evaluate_size=5)
+evaluator.ingest(retrievals=[retrieval], db=db, ingest_size=20)
+
+yield evaluator
+remove_path(bm25_path, pickle_path)
+```
+As you can see, `generate_path` function generate bm25 path and pikle path
+matched dataset name. And, `set_beir_evaluator` function generate
+`retrieval`, `db`, `llm`, `pipeline`. If you want to modify them, you can
+set others in `set_beir_evaluator`.
```

Comment:
You didn't make generate_path and set_beir_evaluator function for BeirEvaluator. Usage of BeirEvaluator is literally same with other evaluator, right?
You dind't read this AI generated documentation, right?